speaking-portal-project/src/phonemeFactory/rhubarb.ts

```diff
@@ -35,26 +35,23 @@ export async function rhubarbProcessor(
         '-d',
         `.${text_file_name}`,
         '--extendedShapes',
-        'GX',
+        'GHX',
         `.${audio_file_name}`,
     ]
 
     const rhubarbProc = spawnSync('./rhubarb', args)
 
-    if (rhubarbProc.stderr) {
```

Comment:
This is where I meant to put the code from the other comment. I'll just copy and paste it here...

Should we have more of a log when the exit code !=0?

I was looking at the different things we can get out of it, and although we might not want all of these to console, they're possible things to add!

And do we want to throw the `stderr` contents or the `error` object?

```
    if (rhubarbProc.status != 0){ // exit code of 0 means it worked successfully
        console.log('exit code (status): ', rhubarbProc.status) // exit code of child process
        console.log('stdout is: ', rhubarbProc.stdout) // contents of output[1]
        console.log('stderr is ', rhubarbProc.stderr) // contents of output[2]
        console.log('signal is ', rhubarbProc.signal) // signal used to kill the child process
        throw Error(`${rhubarbProc.error}`) // error object if child process failed or timed out
    }
```

Reply:
Not sure if the same can be said for ffmpeg, but it seems the only error response we're getting from rhubarb is within the stderr variable, exit code could be helpful to print but everything else seems to be undefined